url4192078a2f_unsegnor_hipotetizador_tgz-pJ8-hipotetizador_MedirTiemposJ8/src/hipotetizador/Hipo.java

```diff
@@ -0,0 +1,1790 @@
+                }
+            }
+            //Si hay alguna regla con subíndice 0 y dispara entonces dispara
+            dispara = dispara && alguna;
+            //Si se dispara comprobamos el consecuente y lo escribimos en la respuesta
+            if (dispara) {
+                for (Elemento e : r.getConsecuente().getElementos()) {
+                    //TODO cambiar esto
+                    //Comprobamos sólo los consecuentes de la segunda línea
+
+                    if (e.getSubindice() == 1) {
+                        //Lo imprimimos en la respuesta ()
+                        //TODO habría que comprobar si ya se había impreso otra cosa y hay conflicto entre las reglas, suponemos que no lo hay
+                        //respuesta[e.getEntrada()] = e.isVerdadero();
+                        contadores[e.getEntrada()]++; //Sumamos uno a los que han contribuido al resultado
+                        if (e.isVerdadero()) {
+                            resultado[e.getEntrada()]++; //Si es verdadero sumamos un voto
+                        } else {
+                            resultado[e.getEntrada()]--; //Si es falso restamos un voto
+                        }
+                    }
+
+                }
+            }
+        }
+
+        //La respuesta se escribe si estaban todos de acuerdo, sino queda como incógnita
+        //0->false, 1->true, 2->hay contradicciones, 3->no hay reglas
+
+        for (int i = 0; i < b.length; i++) {
+            if (contadores[i] > 0) {
+                if (resultado[i] == contadores[i]) {
+                    //entonces estaban todas las reglas de acuerdo y es verdadero
+                    respuesta[i] = 1;
+                } else if (resultado[i] == -contadores[i]) {
+                    //Si el resultado es igual al negativo de los contadores es que
+                    //todas las reglas estaban de acuerdo y es falso
+                    respuesta[i] = 0;
+                } else {
+                    //En otro caso es que existen contradicciones entre las reglas
+                    respuesta[i] = 2;
+                    //Lo hacemos por votación
+                    if (resultado[i] > 0) {
+                        respuesta[i] = 1;
+                    } else if (resultado[i] < 0) {
+                        respuesta[i] = 0;
+                    }
+
+                }
+            } else {
+                //No se han disparado reglas para esta variable
+                respuesta[i] = 3;
+            }
+        }
+
+        return respuesta;
+    }
+
+    private float[][] evaluar_probabilidad(boolean[][] entradas, Teoria teoria) {
+        throw new UnsupportedOperationException("Not yet implemented");
+    }
+
+    private float[][] evaluar_probabilidad(int[][] entradas, Teoria teoria) {
+        return evaluar_probabilidad(entradas, teoria);
+    }
+
+    //Rellena las entradas que faltan según las reglas
+    private float[][] evaluar_probabilidad(float[][] entradas, Teoria teoria) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 25. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.